File: backend/schemas/theme_pulse.py
"""테마 펄스 API 스키마."""

from pydantic import BaseModel
from typing import Optional


class TopStock(BaseModel):
    code: str
    name: str
    news_count: int


class ThemePulseItem(BaseModel):
    rank: int
    theme_name: str
    news_count: int
    high_importance_count: int
    momentum: float
    catalyst_types: dict[str, int]
    top_stocks: list[TopStock]
    setup_score: float = 0.0
    setup_rank: Optional[int] = None


class ThemePulseResponse(BaseModel):
    items: list[ThemePulseItem]
    total_themes: int
    total_news: int
    period_days: int
    generated_at: str


class TimelineDataPoint(BaseModel):
    date: str
    count: int


class TimelineTheme(BaseModel):
    name: str
    data: list[TimelineDataPoint]


class TimelineResponse(BaseModel):
    dates: list[str]
    themes: list[TimelineTheme]
    generated_at: str


class CatalystDistItem(BaseModel):
    type: str
    count: int
    ratio: float


class ImportanceDistItem(BaseModel):
    level: str
    count: int
    ratio: float


class CatalystDistributionResponse(BaseModel):
    catalyst_distribution: list[CatalystDistItem]
    importance_distribution: list[ImportanceDistItem]
    total_news: int
    period_days: int
    generated_at: str

File: frontend/src/types/theme_pulse.ts
export interface TopStock {
  code: string
  name: string
  news_count: number
}

export interface ThemePulseItem {
  rank: number
  theme_name: string
  news_count: number
  high_importance_count: number
  momentum: number
  catalyst_types: Record<string, number>
  top_stocks: TopStock[]
  setup_score: number
  setup_rank: number | null
}

export interface ThemePulseResponse {
  items: ThemePulseItem[]
  total_themes: number
  total_news: number
  period_days: number
  generated_at: string
}

export interface TimelineDataPoint {
  date: string
  count: number
}

export interface TimelineTheme {
  name: string
  data: TimelineDataPoint[]
}

export interface TimelineResponse {
  dates: string[]
  themes: TimelineTheme[]
  generated_at: string
}

export interface CatalystDistItem {
  type: string
  count: number
  ratio: number
}

export interface ImportanceDistItem {
  level: string
  count: number
  ratio: number
}

export interface CatalystDistributionResponse {
  catalyst_distribution: CatalystDistItem[]
  importance_distribution: ImportanceDistItem[]
  total_news: number
  period_days: number
  generated_at: string
}
